### PR TITLE
New boilerplate parser - WIP

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/builder/generation/parsing/members/component.dart [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      node.hasAnnotationWithNames({'Component2', 'AbstractComponent2'});
+  bool isComponent2(Version version) =>
+      version == Version.v4_mixinBased || hasComponent2OrAbstractAnnotation;
+
+  @override
+  void validate(Version version, ErrorCollector errorCollector) {
+    switch (version) {
+      case Version.v4_mixinBased:
+        final superclass = nodeHelper.superclass;
+        if (const ['UiComponent', 'UiStatefulComponent', 'FluxUiComponent']
```
Comment (greglittlefield-wf) [inline lib/src/builder/generation/parsing/members/component.dart]:
```suggestion
        if (const ['UiComponent', 'UiStatefulComponent', 'FluxUiComponent', 'FluxUiStatefulComponent']
```

File: example/builder/src/basic.over_react.g.dart [hunk 2]
```diff
@@ -158,10 +52,12 @@ abstract class _$$BasicProps extends _$BasicProps
 
   /// The default namespace for the prop getters/setters generated for this class.
   @override
-  String get propKeyNamespace => 'BasicProps.';
+  String get propKeyNamespace => '';
```
Comment (greglittlefield-wf) [inline example/builder/src/basic.over_react.g.dart]: FIXME @greglittlefield-wf this should remain for shorthand syntax. Also maybe we should deprecate this?

**Comment by greglittlefield-wf** [comment]
Closing in favor of https://github.com/Workiva/over_react/pull/481

**Pull request closed without merging**